source/core/shape.cc

```diff
@@ -609,6 +761,89 @@ RBFShape<dim>::initialize_bounding_box()
                                                   Tensor<1, 3>());
 }
 
+template <int dim>
+void
+RBFShape<dim>::determine_likely_nodes_for_one_cell(
+  const typename DoFHandler<dim>::active_cell_iterator &cell,
+  const Point<dim>                                      support_point)
+{
+  if (!cell->is_locally_owned())
+    return;
+  // We exit the function immediately if the cell is already in the map
+  if (likely_nodes_map.find(cell) != likely_nodes_map.end())
+    return;
+
+  bool parent_found  = false;
+  minimal_mesh_level = std::min(minimal_mesh_level, cell->level());
+  std::vector<size_t> temporary_iterable_nodes;
+  if (cell->level() > minimal_mesh_level)
+    try
+      {
+        const auto cell_parent     = cell->parent();
+        const auto parent_iterator = likely_nodes_map.find(cell_parent);
+        if (parent_iterator != likely_nodes_map.end())
+          {
+            parent_found = true;
+            temporary_iterable_nodes.swap(iterable_nodes);
+            iterable_nodes.swap(parent_iterator->second);
+          }
+      }
```

Comment:
If I understand correctly, you check if the cell has a parent, and if it has a parent, you use the parent cell to find the iterable nodes for this cell. Then you check which iterable node applied to the new cell and only keeps those has your iterable nodes for the cell. 

This will be fast only if you start with a coarse mesh and then refine it on a single core. But the problem is that when you refine, you are not sure that the new locally owned cell matches the previous cells set before the refinement. In most cases, it won't help you a lot since the new processor does not necessarily have the relevant information from the parent cell.

I don't know if starting from level 0 of the mesh to find the relevant iterable nodes would not be significantly faster. since you could use the full potential of the tree of the cell. I think it would be possible to use a slightly modified version of find_cells_around_point_with_tree from Lethegridtools.

Tell me if I miss something!

Reply:
No you understood correctly! It could indeed be faster
I'll try it

Reply:
It's much faster